src/Cache/CacheFactory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace SMW\Cache;
+
+use Onoi\Cache\CacheFactory as OnoiCacheFactory;
+use SMW\ApplicationFactory;
+use ObjectCache;
+
+/**
+ * @license GNU GPL v2+
+ * @since 2.2
+ *
+ * @author mwjames
+ */
+class CacheFactory {
+
+	/**
+	 * @since 2.2
+	 *
+	 * @return integer
+	 */
+	public function getMainCacheType() {
+		return ApplicationFactory::getInstance()->getSettings()->get( 'smwgCacheType' );
+	}
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param array $cacheOptions
+	 *
+	 * @return stdClass
+	 */
+	public function newCacheOptions( array $cacheOptions ) {
+		return (object)$cacheOptions;
+	}
+
+	/**
+	 * @since 2.2
+	 *
+	 * @param integer $cacheSize
+	 *
+	 * @return Cache
+	 */
+	public function newFixedInMemoryCache( $cacheSize = 500 ) {
+		return OnoiCacheFactory::getInstance()->newFixedInMemoryCache( $cacheSize );
```

Comment:
`SMW\Cache\Cache` and `SMW\Cache\FixedInMemoryCache` will be removed after the migration is done (`SMW\Cache\Cache` == `Onoi\Cache\Cache`)